src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -596,6 +597,40 @@ private JsValue LoadDocument(JsValue self, JsValue[] args)
                 return LoadDocumentInternal(args[0].AsString());
             }
 
+            private JsValue GetCounter(JsValue self, JsValue[] args)
+            {
+                AssertValidDatabaseContext();
+
+                if (args.Length < 2 || args.Length > 3)
+                    throw new InvalidOperationException($"there is no overload of method 'counter' that takes {args.Length} arguments");
+
+                var signature = args.Length == 2 ? "counter(doc, name)" : "counter(doc, name, raw)";
+
+                if (args[0].IsObject() == false)                
+                    throw new InvalidOperationException($"{signature}: 'doc' must be an object argument");
+                
+                if (!(args[0].AsObject() is BlittableObjectInstance doc))                    
+                    throw new InvalidOperationException($"{signature}: 'doc' is missing 'Id' property");
```

Comment:
That isn't the actual error, you need to be clear what you are looking for, IE, the document instance itself.
Also should allow to pass a string doc id.